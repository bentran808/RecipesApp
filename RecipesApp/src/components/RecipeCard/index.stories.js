import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react-native';
import { recipe } from 'mocks';
import React from 'react';
import CenterView from 'storybook/stories/CenterView';
import RecipeCard from '.';

storiesOf('Recipe Card', module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add('default', () => (
    <RecipeCard
      item={recipe}
      onPressRecipe={action('clicked-card')}
      onPressCart={action('clicked-cart-icon')}
    />
  ));
