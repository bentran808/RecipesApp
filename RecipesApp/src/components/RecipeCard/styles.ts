import { StyleSheet } from 'react-native';
import { width } from 'theme';

const recipeColumns = 2;
// item size
const RECIPE_ITEM_HEIGHT = 150;
const RECIPE_ITEM_MARGIN = 20;
const RECIPE_ITEM_WIDTH = (width - (recipeColumns + 1) * RECIPE_ITEM_MARGIN) / recipeColumns;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginLeft: RECIPE_ITEM_MARGIN,
    marginTop: 20,
    width: RECIPE_ITEM_WIDTH,
    height: RECIPE_ITEM_HEIGHT + 75,
    borderColor: '#ccc',
    borderWidth: 0.5,
    borderRadius: 15
  },
  photo: {
    width: RECIPE_ITEM_WIDTH,
    height: RECIPE_ITEM_HEIGHT,
    borderRadius: 15,
    borderBottomLeftRadius: 0,
    borderBottomRightRadius: 0
  },
  title: {
    flex: 1,
    fontSize: 17,
    fontWeight: 'bold',
    textAlign: 'center',
    color: '#444',
    marginTop: 3,
    marginHorizontal: 5
  },
  category: {
    marginVertical: 5
  }
});

export default styles;
