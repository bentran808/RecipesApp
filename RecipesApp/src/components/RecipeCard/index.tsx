import React, { useCallback } from 'react';
import isEqual from 'react-fast-compare';
import { Image, Text, TouchableOpacity, View } from 'react-native';
import styles from './styles';

interface Props {
  item: Recipe;
  onPressRecipe: (item: Recipe) => void;
}

const RecipeCard = ({ item, onPressRecipe }: Props) => {
  const handlePress = useCallback(() => {
    onPressRecipe(item);
  }, []);

  return (
    <TouchableOpacity onPress={handlePress}>
      <View style={styles.container}>
        <Image style={styles.photo} source={{ uri: item.photo_url }} />
        <Text style={styles.title}>{item.title}</Text>
        <Text style={styles.category}>{item.category.name}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default React.memo(RecipeCard, isEqual);
