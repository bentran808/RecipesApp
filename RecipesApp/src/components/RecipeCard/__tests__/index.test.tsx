import { fireEvent, render } from '@testing-library/react-native';
import RecipeCard from 'components/RecipeCard';
import { recipe } from 'mocks';
import renderer from 'react-test-renderer';

describe('Recipe Card Component', () => {
  const onPressCart = jest.fn();

  test('should render correctly', () => {
    const tree = renderer
      .create(<RecipeCard item={recipe} onPressRecipe={jest.fn()} onPressCart={onPressCart} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should call function onPressCart', () => {
    const { getByTestId } = render(
      <RecipeCard item={recipe} onPressRecipe={jest.fn()} onPressCart={onPressCart} />
    );

    const button = getByTestId('cartBtn');
    fireEvent.press(button);

    expect(onPressCart).toHaveBeenCalled();
  });
});
