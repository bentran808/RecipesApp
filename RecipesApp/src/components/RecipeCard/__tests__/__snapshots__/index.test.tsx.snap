// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe Card Component should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#ccc",
        "borderRadius": 15,
        "borderWidth": 0.5,
        "height": 260,
        "justifyContent": "center",
        "marginLeft": 20,
        "marginTop": 20,
        "width": 345,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://www.texanerin.com/content/uploads/2019/06/nobake-chocolate-cookies-1-650x975.jpg",
        }
      }
      style={
        Object {
          "borderBottomLeftRadius": 0,
          "borderBottomRightRadius": 0,
          "borderRadius": 15,
          "height": 150,
          "width": 345,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#444",
          "flex": 1,
          "fontSize": 17,
          "fontWeight": "bold",
          "marginHorizontal": 5,
          "marginTop": 3,
          "textAlign": "center",
        }
      }
    >
      Oatmeal Cookies
    </Text>
    <Text
      style={
        Object {
          "marginVertical": 5,
        }
      }
    >
      Cookies
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 15,
            "fontWeight": "bold",
          }
        }
      >
        $20
      </Text>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 25,
              "width": 25,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
